djautotask/sync.py

```diff
@@ -62,22 +68,89 @@ def __init__(self):
         self.synced_ids = set()
 
 
+class CustomRequestsTransport(transport.Transport):
```

Comment:
I like how you did this. So I went through the synchronizer's `init_api_connection` method and I see the only time self is used is to reference `request_settings`. I think moving CustomRequestsTransport to its own file and much, or all, of the `init_api_connection` method on to this class is a good idea. That way we can keep more of a sync/api separation like we do in django-connectwise.